src/main/java/io/odpf/firehose/sink/prometheus/PromSink.java

```diff
@@ -0,0 +1,193 @@
+package io.odpf.firehose.sink.prometheus;
+
+import com.gojek.de.stencil.client.StencilClient;
+import com.google.common.annotations.VisibleForTesting;
+import io.odpf.firehose.sink.prometheus.request.PromRequest;
+import com.google.protobuf.DynamicMessage;
+import com.newrelic.api.agent.NewRelic;
+import com.newrelic.api.agent.Trace;
+import cortexpb.Cortex;
+import io.odpf.firehose.consumer.Message;
+import io.odpf.firehose.exception.DeserializerException;
+import io.odpf.firehose.exception.NeedToRetry;
+import io.odpf.firehose.metrics.Instrumentation;
+import io.odpf.firehose.sink.AbstractSink;
+import org.apache.commons.io.IOUtils;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpEntityEnclosingRequestBase;
+import org.apache.http.util.EntityUtils;
+import org.xerial.snappy.Snappy;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+import static io.odpf.firehose.metrics.Metrics.SINK_HTTP_RESPONSE_CODE_TOTAL;
+import static io.odpf.firehose.metrics.Metrics.SINK_MESSAGES_DROP_TOTAL;
+import static io.odpf.firehose.sink.prometheus.PromSinkConstants.SUCCESS_CODE_PATTERN;
+
+/**
+ * the Prometheus Sink. this sink use prometheus remote write api to send data into Cortex.
+ */
+public class PromSink extends AbstractSink {
+
+    private PromRequest request;
+    private HttpEntityEnclosingRequestBase httpRequests;
+    private HttpClient httpClient;
+    private StencilClient stencilClient;
+    private Map<Integer, Boolean> retryStatusCodeRanges;
+    private Map<Integer, Boolean> requestLogStatusCodeRanges;
+
+    /**
+     * Instantiates a new Prometheus sink.
+     *
+     * @param instrumentation            the instrumentation
+     * @param request                    the request
+     * @param httpClient                 the http client
+     * @param stencilClient              the stencil client
+     * @param retryStatusCodeRanges      the retry status code ranges
+     * @param requestLogStatusCodeRanges the request log status code ranges
+     */
+    public PromSink(Instrumentation instrumentation, PromRequest request, HttpClient httpClient, StencilClient stencilClient, Map<Integer, Boolean> retryStatusCodeRanges, Map<Integer, Boolean> requestLogStatusCodeRanges) {
```

Comment:
This code is 90% identical to HTTPSink. Can we create another class like HttpRequestHandler and move the logic in that class and call it in both the sinks?